backend-api/src/functions/asyncActiveSession/tokenService/publicKeyGetter.ts

```diff
@@ -0,0 +1,84 @@
+import {
+  decodeProtectedHeader,
+  importJWK,
+  KeyLike,
+  ProtectedHeaderParameters,
+} from "jose";
+import { errorResult, Result, successResult } from "../../utils/result";
+
+export class PublicKeyGetter implements IPublicKeyGetter {
```

Comment:
We need to move to this class the logic (which currently lives in the `TokenService`) to send a request to STS to get their public keys. If that logic isn't part of this class then we will need to rename `PublicKeyGetter` and the `getPublicKey()` to something else because their names will be misleading. 

But we can merge this PR first and then make any changes in the next.